add docker tutorial for core & nlu #1239

wochinge opened the pull request:
**Proposed changes**:
- see https://github.com/RasaHQ/rasa_core/issues/1239

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] add steps to set up core
- [x] add steps to set up core & nlu
- [x] instructions for custom pipeline
- [x] add steps for a custom action server
- [ ] add steps for duckling
- [x] add steps for custom tracker stores
- [x] updated the changelog

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,539 @@
+of the Docker documentation which describes how to install Docker on a variety
+of operating systems.
+
+Congratulation! 🚀 Now we have got everything set up to get going!
```
The same with using 'we'. I would use 'you' everywhere because it refers to the user directly.  One more thing to this is that I would probably leave the 'Congratulations' part here because it makes it look like the most important part of the tutorial is already finished  :)

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,539 @@
+
+Rasa Core works by learning from example conversations. Therefore, we now need
+to write a few examples to get started.
+Stories are written in `Markdown <https://en.wikipedia.org/wiki/Markdown>`_
```
I would suggest having this part after you explain which directories a user will have to create because there is too much going back and forth. Maybe it would even make sense to create a separate step for preparing the directory and then diving into stories

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,539 @@
+      --stories project/data/stories.md \
+      --out models
+
+Yey! You just trained your chatbot for the first time!
```
I think there are too many excitement words included :) For documentation, I would suggest keeping documentation content a bit more toned down because developers should use the docs as a concise and guide of how to use the software.

JustinaPetr replied:
Also, I think it's better to leave out 'for the first time' part, because the quickstart is the place where we expect people to build their first Rasa assistant.

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,539 @@
+
+  []
+
+Hooray! 🎉 🎊 We can now train Rasa Core, Rasa NLU and speak to our bot in
```
Should this really be about training? I think the training steps are already included above.

wochinge commented:
@JustinaPetr I implemented your comments and added the next steps.
It is still quite long, but I don't know where to shorten it without losing information.
I am also not sure about Duckling since I don't know how common it is to use Duckling as part of the NLU pipeline. Generally I tend to include in the tutorial as then everything is covered.

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,684 @@
+
+.. note::
+
+  Depending on the selected pipeline you might have to use a different
```
Where a user can find more information about that? Would be good to add a link or something, because people do you their own custom configs very often.

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,684 @@
+      dispatcher.utter_message(joke) #send the message back to the user
+      return []
+
+Then add the custom action in your stories and your domain file.
```
'Then' repeats two times in a row. What about alternating it with 'Next'? :)

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,684 @@
+      return []
+
+Then add the custom action in your stories and your domain file.
+Continuing the example from above replace replace ``utter_cheer_up`` in
```
Typo: 'replace' is written two times :)

JustinaPetr commented on docs/docker_walkthrough.rst:
```diff
@@ -0,0 +1,684 @@
+  action_endpoint:
+    url: http://action_server:5055/webhook
+
+If you then run ``docker-compose up`` the action server starts together
```
I would rephrase this as a mandatory step: 'Run 'docker-compose up' to start the action server together with...'.

JustinaPetr commented:
Awesome! I would be keen on leaving the duckling out just because it's very optional unless there are some special configurations which a developer would have to implement for the duckling to work. Is it very different from adding additional components and adding it to the config files?

wochinge commented:
@JustinaPetr Adding Duckling is just as the other components, except that the url of has to be added within the NLU pipeline

wochinge commented:
@JustinaPetr Updated it according to your notes

JustinaPetr commented:
@wochinge Looks good! :)
